native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/OutParameterProcessor.java

```diff
@@ -61,17 +69,152 @@ private OutParameterProcessor() {
     }
 
     public static Object getOutParameterValue(BObject result, BTypedesc typeDesc) {
+        try {
+            populateOutParameter(result);
+        } catch (SQLException | ApplicationError e) {
+            return ErrorGenerator.getSQLError(e, "Failed to read OUT parameter value.");
+        }
+
         return get(result, typeDesc, DefaultResultParameterProcessor.getInstance(), "OutParameter");
     }
 
     public static BStream getOutCursorValue(BObject result, BTypedesc typeDesc) {
+        try {
+            populateOutParameter(result);
+        } catch (SQLException | ApplicationError e) {
+            return getErrorStream(typeDesc,
+                    ErrorGenerator.getSQLError(e, "Failed to read parameter value."));
+        }
         return get(result, typeDesc, DefaultResultParameterProcessor.getInstance());
     }
 
     public static Object getInOutParameterValue(BObject result, BTypedesc typeDesc) {
+        try {
+            populateOutParameter(result);
+        } catch (SQLException | ApplicationError e) {
+            return ErrorGenerator.getSQLError(e, "Failed to read INOUT parameter value.");
+        }
         return get(result, typeDesc, DefaultResultParameterProcessor.getInstance(), "InOutParameter");
     }
 
+    private static void populateOutParameter(BObject parameter) throws SQLException, ApplicationError {
+        int paramIndex = (int) parameter.getNativeData(PARAMETER_INDEX_META_DATA);
+        int sqlType = (int) parameter.getNativeData(Constants.ParameterObject.SQL_TYPE_NATIVE_DATA);
+        CallableStatement statement = (CallableStatement) parameter.getNativeData(STATEMENT_NATIVE_DATA_FIELD);
+        BObject procedureCallResult = (BObject) parameter.getNativeData(PROCEDURE_CALL_RESULT);
+        AbstractResultParameterProcessor resultParameterProcessor = (AbstractResultParameterProcessor) parameter
+                .getNativeData(RESULT_PARAMETER_PROCESSOR);
+        Object result;
+        switch (sqlType) {
```

Comment:
Any reason not to use an [enhanced switch clause](https://openjdk.org/jeps/325) in here?

Reply:
Not really, I have just move the code block from the other file. Will update with the suggestion.

Reply:
Fixed [49dc233](https://github.com/ballerina-platform/module-ballerina-sql/pull/743/commits/49dc2338eacff202cfb7676a37577e3746510a43)